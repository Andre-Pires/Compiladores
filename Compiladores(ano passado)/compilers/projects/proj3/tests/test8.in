t1 = a
t2 = b
t3 = t1 + t2
a = t3
t1 = 0
t2 = 1
b = t1 + t2
t1 = a
t2 = b
t3 = t1 + t2
a = t3
c = 0
t1 = 1
t2 = 2
t3 = 3
t4 = 4
t5 = a
t6 = t1 - t2
t7 = t3 + t4
t8 = t5 + t6
t9 = t7 * t8
a = t9
b = 0
