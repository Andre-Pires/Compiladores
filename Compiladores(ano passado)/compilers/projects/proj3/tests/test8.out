 Number of Registers: 5
	r3 = ARP + 4
	r3 = *r3
	r3 = r3
	r2 = ARP + 8
	r2 = *r2
	r2 = r2
	r1 = r3 + r2
	r0 = r1
	r3 = 0
	r2 = 1
	r1 = r3 + r2
	r3 = r0
	r2 = r1
	r4 = ARP + 4
	*r4 = r0
	r0 = r3 + r2
	r0 = r0
	r4 = ARP + 8
	*r4 = r1
	r1 = 0
	r3 = 1
	r2 = 2
	r4 = ARP + 12
	*r4 = r1
	r1 = 3
	r1 = 4
	r4 = ARP + 4
	*r4 = r0
	r0 = r0
	r2 = r3 - r2
	r3 = r3 + r1
	r0 = r0 + r2
	r0 = r3 *r0
	r0 = r0
	r4 = ARP + 4
	*r4 = r0
	r0 = 0
	r4 = ARP + 8
	*r4 = r0
 Number Load Operations: 2
 Number Store Operations: 6
